### PR TITLE
RTC enabled

### PR DESCRIPTION
**Pull request by askpatrickw**
I looked at the other ports, picked the simplest one that had RTC an copied it.
In the original port it used `tv` in the [espressif docs](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/system/system_time.html#rtc-clock-source) it used `tv_now`.
This is a partial implemenation for #3321 

NOTES:

* I'm not sure if the calibration stuff which was essentially n\a in the one I copied should be implemented for the ESP32S2
* In the [Espessif config](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/kconfig.html#config-esp32s2-rtc-clk-src) you can configure different sources for the RTC to run off. Currently we're using `ESP32S2_RTC_CLK_SRC_INT_RC`. It might be better (but more power hungry) to use `ESP32S2_RTC_CLK_SRC_INT_8MD256` as they claim it has less wander. I'm not 100% sure if that's going to be true with CP today as I believe this has to do with lower power modes.

Seems like its working... Tested on a Kaluga

```python
>>> import time
>>> import rtc
>>> r = rtc.RTC()
>>> r.datetime = time.struct_time((2020, 9, 14, 23, 22, 15, 0, -1, -1))
>>> current_time = r.datetime
>>> print(current_time)
struct_time(tm_year=2020, tm_mon=9, tm_mday=14, tm_hour=23, tm_min=22, tm_sec=25, tm_wday=0, tm_yday=258, tm_isdst=-1)
>>> help("modules")
__main__          errno             neopixel_write    struct
_os               espidf            os                supervisor
_pixelbuf         fontio            pulseio           sys
_time             framebufferio     pwmio             terminalio
array             gamepad           random            time
bitbangio         gc                re                touchio
board             io                rtc               ulab
builtins          ipaddress         sdcardio          usb_hid
busio             json              sharpdisplay      vectorio
collections       math              socketpool        wifi
digitalio         microcontroller   ssl
displayio         micropython       storage
Plus any modules on the filesystem
```

### CODE REVIEW

File: ports/esp32s2/mpconfigport.mk [hunk 1]
```diff
@@ -21,7 +21,7 @@ CIRCUITPY_COUNTIO = 0
 CIRCUITPY_FREQUENCYIO = 0
 CIRCUITPY_I2CPERIPHERAL = 0
 CIRCUITPY_ROTARYIO = 0
-CIRCUITPY_RTC = 0
+CIRCUITPY_RTC = 1
```
Comment (tannewt) [inline ports/esp32s2/mpconfigport.mk:24]: Could you try removing this line instead? I think it will be on by default.

**Review by tannewt — CHANGES_REQUESTED** [review]
One question. Otherwise, this looks good. Thank you!

File: ports/esp32s2/mpconfigport.mk (hunk 1 above)
Comment (askpatrickw) [reply ports/esp32s2/mpconfigport.mk:24]: That did work! I retested on the Kaluga.

Comment (tannewt) [reply ports/esp32s2/mpconfigport.mk:24]: Don't change FULL_BUILD. Delete this line.

**Review by tannewt — CHANGES_REQUESTED** [review]

File: ports/esp32s2/mpconfigport.mk (hunk 1 above)
Comment (askpatrickw) [reply ports/esp32s2/mpconfigport.mk:24]: OK.. moving slower. I reverted that change. 

With CIRCUITPY_RTC = 0 no RTC is available in the image so I think the original PR was the correct change. If I'm not understanding let me know.

Comment (askpatrickw) [reply ports/esp32s2/mpconfigport.mk]: @tannewt  Yes, when I remove the CIRCUITPY_RTC line completely RTC is present in the build.

**Review by tannewt — APPROVED** [review]
Great! Thank you!

**Comment by anecdata** [comment]
fwiw, calibration is enabled on SAMD51, but it's a small adjustment relative to some typical drifts. For example, my PyPortal gains about a second per minute, well outside of the calibration range.

**Pull request merged**